frame/staking/src/mock.rs

```diff
@@ -503,13 +503,21 @@ fn post_conditions() {
 }
 
 fn check_count() {
-	let nominator_count = Nominators::<Test>::iter().count() as u32;
-	let validator_count = Validators::<Test>::iter().count() as u32;
-	assert_eq!(nominator_count, CounterForNominators::<Test>::get());
-	assert_eq!(validator_count, CounterForValidators::<Test>::get());
+	// @kianenigma
+	// TODO: checking this is good for tests of top level staking api, but when
+	// unit testing parts of voter_bags we can't expect this to update properly
+	// (unless we manually do it in the test, which might be ok?).
+	// Instead I think the debug_asserts should be enough?
+	// Otherwise maybe we can have a `build_and_execute_without_check_count`, but that
```

Comment:
`build_and_execute_without_check_count` sounds good, but use it only with extra attention, and it should ONLY be used in the `VoterList` api later or lower, not the top level tests.